spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/convert/DefaultDatastoreEntityConverter.java

```diff
@@ -122,7 +127,39 @@ public ReadWriteConversions getConversions() {
 			throw new DatastoreDataException("Unable to read " + persistentEntity.getName() + " entity", ex);
 		}
 
-		return instance;
+		return (R) instance;
+	}
+
+	private DatastorePersistentEntity getDiscriminationPersistentEntity(DatastorePersistentEntity ostensibleEntity,
+			EntityPropertyValueProvider propertyValueProvider) {
+		if (ostensibleEntity.getDiscriminationFieldName() == null) {
+			return ostensibleEntity;
+		}
+		DatastorePersistentEntity persistentEntity = null;
+
+		Set<Class> members = DatastoreMappingContext.getDiscriminationFamily(ostensibleEntity.getType());
+
+		if (members != null) {
+			for (Class member : members) {
```

Comment:
is it possible to replace this loop with a stream?

Reply:
I tried but a stream using an anonymous function doesn't work here because we need to set the `persistentEntity` variable inside the loop and anon functions can't set anything outside (it requires variables coming in to be final).